### PR TITLE
Drop deprecated action APIs (T797)

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 aaand another

File: include/crm/common/actions.h [hunk 1]
```diff
@@ -16,13 +16,7 @@
 #include <libxml/tree.h>                // xmlNode
 
 #include <crm/lrmd_events.h>            // lrmd_event_data_t
-
-#include <glib.h>                       // GList, GHashTable
```
Comment (nrwahl2) [inline include/crm/common/actions.h:20]: We still have at least one `gboolean`

File: include/crm/common/actions.h [hunk 2]
```diff
@@ -16,13 +16,7 @@
...
-#include <libxml/tree.h>                // xmlNode
```
Comment (nrwahl2) [inline include/crm/common/actions.h:21]: still have `xmlNode`s

File: include/crm/common/actions_internal.h [hunk 3]
```diff
@@ -119,6 +122,51 @@ enum pcmk__action_type {
...
+    enum rsc_start_requirement needs;   // Prerequisite for recovery
+    enum action_fail_response on_fail;  // Response to failure
+    enum rsc_role_e fail_role;          // Resource role if action fails
+    GHashTable *meta;                   // Meta-attributes relevant to action
+    GHashTable *extra;                  // Action-specific instance attributes
+
+    /* Current count of runnable instance actions for "first" action in an
+     * ordering dependency with pcmk__ar_min_runnable set.
+     */
+    int runnable_before;                // For Pacemaker use only
```
Comment (nrwahl2) [inline include/crm/common/actions_internal.h:234]: Can drop "For Pacemaker use only" now

File: lib/common/actions.c [hunk 4]
```diff
@@ -144,37 +144,37 @@ pcmk__parse_action(const char *action_name)
  * \return Static string describing \p on_fail
  */
 const char *
-pcmk__on_fail_text(enum action_fail_response on_fail)
+pcmk__on_fail_text(enum pcmk__on_fail on_fail)
```
Comment (nrwahl2) [inline lib/common/actions.c:147]: Not new but this function should also have `\internal` in doxygen

**Review by nrwahl2 — COMMENTED** [review]
Looking good

File: lib/common/actions.c [hunk 5]
```diff
@@ -89,7 +89,7 @@ pcmk_action_text(enum action_tasks action)
  * \return Action type corresponding to \p action_name
  */
 enum action_tasks
-pcmk_parse_action(const char *action_name)
+pcmk__parse_action(const char *action_name)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/actions.c:92]: Ah bummer, thanks

File: lib/common/actions.c [hunk 6]
```diff
@@ -34,7 +34,7 @@
  * \return Static string describing \p action
  */
 const char *
-pcmk_action_text(enum action_tasks action)
+pcmk__action_text(enum action_tasks action)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/actions.c]: thx

File: lib/common/actions.c [hunk 7]
```diff
@@ -144,7 +144,7 @@ pcmk__parse_action(const char *action_name)
  * \return Static string describing \p on_fail
  */
 const char *
-pcmk_on_fail_text(enum action_fail_response on_fail)
+pcmk__on_fail_text(enum action_fail_response on_fail)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/actions.c]: thx

File: lib/pacemaker/pcmk_simulate.c [hunk 8]
```diff
@@ -288,12 +288,12 @@ write_sim_dotfile(pcmk_scheduler_t *scheduler, const char *dot_file,
...
                 optional = false;
                 style = PCMK__VALUE_BOLD;
-            } else if ((uint32_t) before->type == pcmk__ar_none) {
+            } else if (before->flags == pcmk__ar_none) {
                 continue;
             } else if (pcmk_is_set(before->action->flags,
                                    pcmk_action_added_to_graph)
                        && pcmk_is_set(action->flags, pcmk_action_added_to_graph)
-                       && (uint32_t) before->type != pcmk__ar_if_on_same_node_or_target) {
+                       && before->flags != pcmk__ar_if_on_same_node_or_target) {
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pacemaker/pcmk_simulate.c:296]: would push past 80 characters, I figured this was the lesser evil :)

File: include/crm/common/actions.h (hunk 2 above)
Comment (kgaillot) [reply include/crm/common/actions.h:21]: it was included twice :)

File: lib/common/actions.c (hunk 4 above)
Comment (kgaillot) [reply lib/common/actions.c:147]: thx

File: lib/pacemaker/pcmk_sched_actions.c [hunk 9]
```diff
@@ -464,7 +464,7 @@ update_action_for_ordering_flags(pcmk_action_t *first, pcmk_action_t *then,
         // Don't bother marking 'first' as changed just for this
     }
 
-    if (pcmk_any_flags_set(order->type, pcmk__ar_first_implies_then
+    if (pcmk_any_flags_set(order->flags, pcmk__ar_first_implies_then
                                         |pcmk__ar_then_implies_first
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:468]: thx

File: lib/pacemaker/pcmk_sched_actions.c [hunk 10]
```diff
@@ -1383,7 +1383,7 @@ pcmk__deduplicate_action_inputs(pcmk_action_t *action)
              * matter, but crm_simulate looks at certain ones when creating a
              * dot graph. Combining the flags is sufficient for that purpose.
              */
-            last_input->type |= input->type;
+            last_input->flags |= input->flags;
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:1386]: makes sense

File: include/crm/common/actions.h (hunk 1 above)
Comment (kgaillot) [reply include/crm/common/actions.h:20]: it was included twice too :)

**Comment by kgaillot** [comment]
updated per review

**Comment by kgaillot** [comment]
CI issues are not relevant (coverity redetected the Y2038 issues since this hasn't been rebased)

File: lib/pacemaker/pcmk_simulate.c (hunk 8 above)
[the remark this answers is not in the document]
Comment (nrwahl2) [reply lib/pacemaker/pcmk_simulate.c:296]: Makes sense. Could rearrange some lines but no need to bother especially right now.

File: include/crm/common/actions.h (hunk 2 above)
Comment (nrwahl2) [reply include/crm/common/actions.h:21]: ah, sure was. The diff wasn't showing the lines above these by default.

File: lib/common/actions.c (hunk 4 above)
Comment (nrwahl2) [reply lib/common/actions.c:150]: ah I commented on this twice. GitHub didn't show the earlier comment when I made this one, so I assumed this was an old function the second time.

**Review by nrwahl2 — APPROVED** [review]

**Pull request merged**